app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java

```diff
@@ -599,4 +599,16 @@ public static void playWithKore(final Context context, final Uri videoURL) {
         intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
         context.startActivity(intent);
     }
+
+    /**
+     * Finish provided <code>Activity</code> and then start <code>MainActivity</code>.
```

Comment:
not just the provided activity is finished, all activities sharing the same task. As NewPipe has the Manifest flag "singleTask", all activies are finished. SingleTask is set here: https://github.com/TeamNewPipe/NewPipe/blob/d6e0bd8c26af1b7818e025d8dcda2cb69f636fab/app/src/main/AndroidManifest.xml#L31

https://developer.android.com/reference/android/app/Activity#finishAffinity()
"Finish this activity as well as all activities immediately below it in the current task that have the same affinity."

Reply:
Okay, would the following verbiage work?

"Finish this activity as well as all activities running below it and then start MainActivity".

Reply:
sounds good to me